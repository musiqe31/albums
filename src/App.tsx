import { BrowserRouter as Router, Route, Routes } from 'react-router-dom'
import Dashboard from './components/Dashboard/dashboard'
import Checkout from './components/Checkout/checkout'
import Header from './components/Header/header'
import NotFound from './components/ErrorPage/errorPage'
import './App.css'
import { useEffect } from 'react'
import { useDispatch } from 'react-redux'
import { getAllAlbums } from './Redux/albumSlice'
import { api } from './messages'

const App = () => {
  const dispatch = useDispatch()
  
  useEffect(() => {
    const getData = async () => {
      const response = await fetch(api.url)
      const data = await response.json()
      dispatch(getAllAlbums(data))
    }
    getData()
  }, [])

  return (
      <Router>
        <Header />
        <Routes>
          <Route path="/" element={<Dashboard />} />
          <Route path="/checkout" element={<Checkout />} />
          <Route path="*" element={<NotFound />} />
        </Routes>
      </Router>
  )
}

export default App
