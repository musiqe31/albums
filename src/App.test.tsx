import React from 'react';
import { render } from '@testing-library/react';
import App from './App';
import { Provider } from 'react-redux';
import Store from './Redux/store';

describe('App', () => {
  it('renders Item title', () => {
    const app = render(
      <Provider store={Store}>
        <App />
      </Provider>
    );
    expect(app).toMatchSnapshot();
  });
})