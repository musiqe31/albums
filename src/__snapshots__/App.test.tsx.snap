// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders Item title 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="navbar"
      >
        My Album Store
        <a
          href="/"
        >
          Dashboard
        </a>
        <a
          href="/checkout"
        >
          Checkout
        </a>
        Cart Total :
         
        0
      </div>
      <div
        class="itemContainer"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="navbar"
    >
      My Album Store
      <a
        href="/"
      >
        Dashboard
      </a>
      <a
        href="/checkout"
      >
        Checkout
      </a>
      Cart Total :
       
      0
    </div>
    <div
      class="itemContainer"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Item initiates add to cart function 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="soloAlbum"
        style="border-color: black;"
      >
        <p>
          33
        </p>
        <p
          data-testid="title"
        >
          This is a test
        </p>
        <div
          class="addorremove"
        >
          <button
            data-testid="add"
            style="font-size: 30px;"
          >
            ✅
          </button>
          <button
            data-testid="remove"
            disabled=""
            style="font-size: 30px;"
          >
            ❌
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="soloAlbum"
      style="border-color: black;"
    >
      <p>
        33
      </p>
      <p
        data-testid="title"
      >
        This is a test
      </p>
      <div
        class="addorremove"
      >
        <button
          data-testid="add"
          style="font-size: 30px;"
        >
          ✅
        </button>
        <button
          data-testid="remove"
          disabled=""
          style="font-size: 30px;"
        >
          ❌
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
